File: Java/백준/Silver/15828. Router/Router.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringBuilder sb = new StringBuilder();

        Queue<Integer> router = new LinkedList<>();
        int n = Integer.parseInt(br.readLine());

        while (true) {
            int packet = Integer.parseInt(br.readLine());

            if (packet == -1) {
                break;
            }

            else if (packet == 0) {
                router.poll();
            }

            else {
                if (router.size() >= n){
                    continue;
                }
                else{
                    router.add(packet);
                }
            }
        }

        if (router.isEmpty()) {
            sb.append("empty");
        } else {
            while (!router.isEmpty()) {
                sb.append(router.poll()).append(" ");
            }
        }

        bw.write(sb.toString().trim());
        bw.flush();
    }
}

File: Python/백준/Silver/15828. Router/Router.py
import sys
from collections import deque
input = sys.stdin.readline

router = deque()
n = int(input())

while True:
    packet = int(input())

    if packet == -1:
        break

    elif packet == 0:
        router.popleft()

    else:
        if len(router) >= n:
            continue
        else:
            router.append(packet)
print(" ".join(map(str, router)) if router else "empty")